src/App.js

```diff
@@ -205,7 +206,9 @@ export default class App extends Base {
                 Keyboard.provide('xrn:firebolt:capability:input:keyboard', new KeyboardProviderDelegater(this));
                 PinChallenge.provide('xrn:firebolt:capability:usergrant:pinchallenge', new PinChallengeProviderDelegater(this));
               } else {
-                Discovery.provide('xrn:firebolt:capability:discovery:interest', new UserInterestDelegater(this));
+                if (process.env.REGISTERPROVIDER) {
```

Comment:
The changes looks good.
However, we would need to implement the same logic for SystemUI provider as well, although in a new ticket.